publish.js

```diff
@@ -662,98 +680,105 @@ exports.publish = function(taffyData, opts, tutorials) {
 
     attachModuleSymbols( find({ longname: {left: 'module:'} }), members.modules );
 
-    // generate the pretty-printed source files first so other pages can link to them
-    if (outputSourceFiles) {
-        generateSourceFiles(sourceFiles, opts.encoding);
-    }
+    view.versions = [];
+    gitSemverTags().then(function(versions){
+        view.versions = versions;
```

Comment:
The changes below aren’t that huge. I just had to move all the file generation methods in here, because getting Git tags is happening asynchronous.

Anyway. I’m not that happy with the amount of changes and it makes me feel like »3. Generate the docs in an Iframe« might be the cleaner way to handle switching versions. 

So I’m thinking about another pull request which implements option 3 like described over here:
https://github.com/nhnent/tui.jsdoc-template/issues/5

I guess I could take care of that tomorrow.